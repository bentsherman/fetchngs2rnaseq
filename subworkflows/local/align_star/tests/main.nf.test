nextflow_workflow {

    name "Test Subworkflow ALIGN_STAR"
    script "../main.nf"
    workflow "ALIGN_STAR"
    config "./nextflow.config"

    tag "STAR_GENOMEGENERATE"
    tag "STAR_GENOMEGENERATE_IGENOMES"
    tag "STAR_ALIGN"
    tag "STAR_ALIGN_IGENOMES"
    tag "BAM_SORT_STATS_SAMTOOLS"

    test("star - no igenomes") {

        setup {
            run("STAR_GENOMEGENERATE") {
                script "../../../../modules/nf-core/star/genomegenerate/main.nf"
                process {
                    """
                    input[0] = Channel.of([
                        [ id:'test_fasta' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                    ])
                    input[1] = Channel.of([
                        [ id:'test_gtf' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                    ])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = false

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(file(workflow.out.log_final[0][1]).name).match("without igenomes - log_final") },
                { assert snapshot(file(workflow.out.log_out[0][1]).name).match("without igenomes - log_out") },
                { assert snapshot(workflow.out.bai).match("without igenomes - bai")},
                { assert snapshot(workflow.out.bam).match("without igenomes - bam")},
                { assert snapshot(workflow.out.bam_sorted).match("without igenomes - bam_sorted")},
                { assert snapshot(workflow.out.bam_transcript).match("without igenomes - bam_transcript")},
                { assert snapshot(workflow.out.csi).match("without igenomes - csi")},
                { assert snapshot(workflow.out.log_progress).match("without igenomes - log_progress")},
                { assert snapshot(workflow.out.fastq).match("without igenomes - fastq")},
                { assert snapshot(workflow.out.flagstat).match("without igenomes - flagstat")},
                { assert snapshot(workflow.out.idxstats).match("without igenomes - idxstats")},
                { assert snapshot(workflow.out.orig_bam).match("without igenomes - orig_bam")},
                { assert snapshot(workflow.out.stats).match("without igenomes - stats")},
                { assert snapshot(workflow.out.tab).match("without igenomes - tab")},
                { assert snapshot(workflow.out.versions).match("without igenomes - versions")}
            )
        }
    }
    test("star - with igenomes") {

        setup {
            run("STAR_GENOMEGENERATE_IGENOMES") {
                script "../../../../modules/local/star_genomegenerate_igenomes/main.nf"
                process {
                    """
                    input[0] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)])
                    input[1] = Channel.of([file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true)])
                    """
                }
            }
        }

        when {
            workflow {
                """
                star_ignore_sjdbgtf = false
                seq_platform        = 'illumina'
                seq_center          = false
                is_aws_igenome      = true

                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_2.fastq.gz', checkIfExists: true)
                    ]
                ])
                input[1] = STAR_GENOMEGENERATE_IGENOMES.out.index.map{ [ [id:'star'], it ] }
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = star_ignore_sjdbgtf
                input[4] = seq_platform
                input[5] = seq_center
                input[6] = is_aws_igenome
                input[7] = Channel.of([
                    [ id:'test_fasta' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(file(workflow.out.log_final[0][1]).name).match("with igenomes - log_final") },
                { assert snapshot(file(workflow.out.log_out[0][1]).name).match("with igenomes - log_out") },
                { assert snapshot(workflow.out.bai).match("with igenomes - bai")},
                { assert snapshot(workflow.out.bam).match("with igenomes - bam")},
                { assert snapshot(workflow.out.bam_sorted).match("with igenomes - bam_sorted")},
                { assert snapshot(workflow.out.bam_transcript).match("with igenomes - bam_transcript")},
                { assert snapshot(workflow.out.csi).match("with igenomes - csi")},
                { assert snapshot(workflow.out.log_progress).match("with igenomes - log_progress")},
                { assert snapshot(workflow.out.fastq).match("with igenomes - fastq")},
                { assert snapshot(workflow.out.flagstat).match("with igenomes - flagstat")},
                { assert snapshot(workflow.out.idxstats).match("with igenomes - idxstats")},
                { assert snapshot(workflow.out.orig_bam).match("with igenomes - orig_bam")},
                { assert snapshot(workflow.out.stats).match("with igenomes - stats")},
                { assert snapshot(workflow.out.tab).match("with igenomes - tab")},
                { assert snapshot(workflow.out.versions).match("with igenomes - versions")}
            )
        }
    }

}
