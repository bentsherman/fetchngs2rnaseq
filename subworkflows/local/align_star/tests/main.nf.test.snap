{
    "without igenomes - log_progress": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.476234"
    },
    "with igenomes - stats": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,99fb9569a1b39fe38d7bd7d71daa8364"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.123236"
    },
    "with igenomes - flagstat": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.063991"
    },
    "without igenomes - flagstat": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.flagstat:md5,db0e25cd0b37d3030e807846c022199e"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.500434"
    },
    "with igenomes - csi": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.008879"
    },
    "without igenomes - stats": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.stats:md5,99fb9569a1b39fe38d7bd7d71daa8364"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.539089"
    },
    "with igenomes - bam_sorted": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b6cec9bba6b04b9b92eddbef128bdfbb"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:34.968826"
    },
    "without igenomes - versions": {
        "content": [
            [
                "versions.yml:md5,35c1cc21d26d9d8f00e6ec87e97fb634",
                "versions.yml:md5,c156b12646d8a51a4d59c42f8d833305",
                "versions.yml:md5,d18a2907f84607859a6723fa33485378",
                "versions.yml:md5,e902002dd55f6cca09ea62c9aa42ef0d",
                "versions.yml:md5,f6b7635a2693b70fe6836c8e45ebba98",
                "versions.yml:md5,f985768ea0cc88e2b533cc08538f3771"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.575417"
    },
    "without igenomes - bam_transcript": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.453855"
    },
    "without igenomes - idxstats": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.511992"
    },
    "with igenomes - fastq": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.046516"
    },
    "without igenomes - log_out": {
        "content": [
            "test.Log.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:07:26.212398"
    },
    "without igenomes - fastq": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.488954"
    },
    "without igenomes - csi": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.464985"
    },
    "with igenomes - versions": {
        "content": [
            [
                "versions.yml:md5,c156b12646d8a51a4d59c42f8d833305",
                "versions.yml:md5,d18a2907f84607859a6723fa33485378",
                "versions.yml:md5,e3b043ba840d5bddde92f0bf1c91a5fc",
                "versions.yml:md5,e902002dd55f6cca09ea62c9aa42ef0d",
                "versions.yml:md5,f6b7635a2693b70fe6836c8e45ebba98",
                "versions.yml:md5,f985768ea0cc88e2b533cc08538f3771"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.173761"
    },
    "with igenomes - log_progress": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Log.progress.out:md5,b2bd061d6cbaaf3d6d3b1fed547f69b8"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.028932"
    },
    "with igenomes - bam_transcript": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:34.989058"
    },
    "with igenomes - log_out": {
        "content": [
            "test.Log.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:04:22.59113"
    },
    "without igenomes - tab": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.557135"
    },
    "without igenomes - bai": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,25b78d2e41dc31350004b7ab9e5c3618"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T17:02:42.995047"
    },
    "without igenomes - orig_bam": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b9ee1c607e07323bc1652ef3babb543f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.523368"
    },
    "without igenomes - log_final": {
        "content": [
            "test.Log.final.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:07:26.199"
    },
    "without igenomes - bam": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,f75b29383d9c16a466ae22dc676b3022"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T17:02:43.076908"
    },
    "with igenomes - orig_bam": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b6cec9bba6b04b9b92eddbef128bdfbb"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.100886"
    },
    "with igenomes - idxstats": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.idxstats:md5,107ca94dd426cc44db316f0d402307c5"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.080603"
    },
    "with igenomes - tab": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.SJ.out.tab:md5,844af19ab0fc8cd9a3f75228445aca0d"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:00:35.148353"
    },
    "with igenomes - bam": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam:md5,37c080a0d854fffee57360f2176670a8"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T17:03:38.323083"
    },
    "without igenomes - bam_sorted": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.Aligned.sortedByCoord.out.bam:md5,b9ee1c607e07323bc1652ef3babb543f"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T11:59:42.442171"
    },
    "with igenomes - bai": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bam.bai:md5,25b78d2e41dc31350004b7ab9e5c3618"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T17:03:38.264526"
    },
    "with igenomes - log_final": {
        "content": [
            "test.Log.final.out"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-07T12:04:22.57863"
    }
}